### DISCUSSION TITLE
3D circular button of an image

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#6090] [Digital Image Processing]
Centos 8.3
Imagemagick 7+

How do I create a raised button of an image like the one below with shining light?

![cropped_photo](https://user-images.githubusercontent.com/12046302/220220754-3d045a56-35f5-4d12-b755-479851dc7a0c.png)

**Comment by fmw42** [comment]
Can you show your desired output. What is "shining light"? Your image is not raised. So was that the input or output?

**Reply by srini8080** [reply]
![californiabear](https://user-images.githubusercontent.com/12046302/220285845-c3bafffa-85e7-4467-ae4a-34ad676617da.jpg)

the above turned into

![californiabear3d](https://user-images.githubusercontent.com/12046302/220285859-1362b898-a2f8-4cc2-96a0-8310e90fd630.jpg)

something like this

**Comment by fmw42** [comment]
Here is one approach which follows at https://imagemagick.org/Usage/thumbnails/#badge_paint_mask

```
Line 1: set input image
Line 2: set blur for raised button
Line 3: read the input, set the dimensions (dim) to the min of width and height, set the filename
Line 4-5: create a white circle on a black background the size of dim
Line 6: blur the circle image
Line 7: create black mask image with transparency on the outside
Line 8-9: create a black image and put the negated blurred circle in the alpha channel as the second blurred circle
Line 10-12: combine the input, blurred circle with alpha and the mask to create the output

```
<br>

Unix Syntax:

```
infile="220285845-c3bafffa-85e7-4467-ae4a-34ad676617da.jpg"
blur=40      
magick "$infile" -set option:dim "%[fx:min(w,h)]" -set filename:f "%t" -write mpr:img +delete \
\( -size "%[dim]x%[dim]" xc:black -fill white -draw "circle %[fx:dim/2],%[fx:dim/2] %[fx:dim-$blur/2],%[fx:dim/2]" \
    -alpha off -write mpr:circle +delete \) \
\( mpr:circle -blur 0x$blur -write mpr:circle_blur +delete \) \
\( mpr:circle -negate mpr:circle -alpha off -compose copy_opacity -composite -write mpr:circle_mask +delete \) \
\( mpr:circle_blur -fill black -colorize 100 \( mpr:circle_blur -negate \) \
    -alpha off -compose copy_opacity -composite -write mpr:circle_blur2 +delete \) \
mpr:img -alpha set -gravity center -compose over -extent "%[dim]x%[dim]"  \
mpr:circle_blur2 -composite mpr:circle_mask -compose dstin -composite \
"%[filename:f]_button.png"

```

![220285845-c3bafffa-85e7-4467-ae4a-34ad676617da_button](https://user-images.githubusercontent.com/13458218/220438015-7d39f895-f484-417c-9b37-ba6cc3302a35.png)

**Reply by srini8080** [reply]
That worked. Thank you.
